engi.go

```diff
@@ -25,32 +21,53 @@ var (
 	scaleOnResize   = false
 	fpsLimit        = 120
 	headless        = false
+	vsync           = true
 	resetLoopTicker = make(chan bool, 1)
 )
 
-func Open(title string, width, height int, fullscreen bool, defaultScene Scene) {
-	keyStates = make(map[Key]bool)
-	Time = NewClock()
-	Files = NewLoader()
+type RunOptions struct {
+	// NoRun indicates the Open function should return immediately, without looping
+	NoRun bool
 
-	run(defaultScene, title, width, height, fullscreen)
-}
+	// Title is the Window title
+	Title string
 
-func OpenHeadless(defaultScene Scene) {
-	keyStates = make(map[Key]bool)
-	Time = NewClock()
-	Files = NewLoader() // TODO: do we want files in Headless mode?
+	// HeadlessMode indicates whether or not OpenGL calls should be made
+	HeadlessMode bool
 
-	headless = true
+	Fullscreen    bool
+	Width, Height int
 
-	runHeadless(defaultScene)
-}
+	// VSync indicates whether or not OpenGL should wait for the monitor to swp the buffers
+	VSync bool
```

Comment:
Should be changed to `DisableVSync`, because this defaults to `false`, and the default behavior should be to honor vsync.